persistence/trees/trees.go

```diff
@@ -71,26 +71,29 @@ type stateTree struct {
 	nodeStore kvstore.KVStore
 }
 
-var _ modules.TreeStoreModule = &treeStore{}
+var _ modules.TreeStoreModule = &TreeStore{}
 
-// treeStore stores a set of merkle trees that
+// TreeStore stores a set of merkle trees that
 // it manages. It fulfills the modules.TreeStore interface.
 // * It is responsible for atomic commit or rollback behavior
 // of the underlying trees by utilizing the lazy loading
 // functionality provided by the underlying smt library.
-type treeStore struct {
+type TreeStore struct {
 	base_modules.IntegrableModule
 
-	logger       *modules.Logger
-	treeStoreDir string
+	logger *modules.Logger
+	Bus    modules.Bus
```

Comment:
This feels like a code smell. I don't think `Bus` should ever be an exposed structure. It's the "bus" that everyone has access to internally